### PR TITLE
Scenario diffcheck added query parameter unexpected

### PR DESCRIPTION
**Pull request by stollgr**

### CODE REVIEW

File: src/main/java/com/smartsheet/wiremock/extensions/ApiScenarioTransformer.java [hunk 1]
```diff
@@ -158,10 +157,11 @@ static String forMessage(String message, Integer errorCode) {
...
 		private static String getDiff(Request request, JsonNode scenario) {
-			return diffHeaders(request, scenario) + " " +
-					diffUrl(request, scenario) + " " +
-					diffQueryParams(request, scenario) + " " +
-					diffBody(request, scenario).trim();
+			return String.format("%s %s %s %s",
+					diffHeaders(request, scenario),
+					diffUrl(request, scenario),
+					diffQueryParams(request, scenario),
+					diffBody(request, scenario).trim());
```
Comment (cameronbowie) [inline src/main/java/com/smartsheet/wiremock/extensions/ApiScenarioTransformer.java]: Should this .trim() be outside of the last paren? (as in, String.format(...).trim())

File: src/main/java/com/smartsheet/wiremock/extensions/ApiScenarioTransformer.java [hunk 2]
```diff
@@ -212,7 +212,7 @@ private static String diffQueryParams(Request request, JsonNode scenario) {
 				return "";
 			}
 
-			return diffExpectedQueryParams(request, scenarioQueryParams); // + diffUnexpectedQueryParams(request, scenarioQueryParams);
+			return String.format("%s %s",diffExpectedQueryParams(request, scenarioQueryParams), diffUnexpectedQueryParams(request, scenarioQueryParams));
```
Comment (cameronbowie) [inline src/main/java/com/smartsheet/wiremock/extensions/ApiScenarioTransformer.java]: Maybe add .trim()?

File: src/main/java/com/smartsheet/wiremock/extensions/ApiScenarioTransformer.java [hunk 3]
```diff
@@ -285,14 +274,48 @@ private static String diffExpectedQueryParams(Request request, JsonNode scenario
...
 
 			return "";
 		}
+
+
+		private static String diffUnexpectedQueryParams(Request request, JsonNode scenarioQueryParams) {
+
+			StringBuilder queryParamDiff = new StringBuilder();
+
+			String requestQueryParamString = request.getUrl().split("\\?")[1];
```
Comment (cameronbowie) [inline src/main/java/com/smartsheet/wiremock/extensions/ApiScenarioTransformer.java]: What happens if split returns a list with a single element? Could you maybe check the array length in the if statement below and then extract the query param string if it exists?

File: src/main/java/com/smartsheet/wiremock/extensions/ApiScenarioTransformer.java [hunk 4]
```diff
@@ -285,14 +274,48 @@ private static String diffExpectedQueryParams(Request request, JsonNode scenario
...
+			if(requestQueryParamString == null || requestQueryParamString.isEmpty()){
+				return "";
+			}
+
+			//Populate list of scenario expected query parameter keys.
+			Iterator<Map.Entry<String, JsonNode>> scenarioQueryParamIterator = scenarioQueryParams.fields();
+			List<String> scenarioQueryParamKeys = new ArrayList<>();
```
Comment (cameronbowie) [inline src/main/java/com/smartsheet/wiremock/extensions/ApiScenarioTransformer.java]: Could you pull this into a function?

File: src/main/java/com/smartsheet/wiremock/extensions/ApiScenarioTransformer.java [hunk 5]
```diff
@@ -285,14 +274,48 @@ private static String diffExpectedQueryParams(Request request, JsonNode scenario
...
+			while (scenarioQueryParamIterator.hasNext()) {
+				Map.Entry<String, JsonNode> scenarioQueryParam = scenarioQueryParamIterator.next();
+				scenarioQueryParamKeys.add(scenarioQueryParam.getKey().trim());
+			}
+
+			//Populate list of query parameter keys found in request.
+			List<String> requestQueryParamKeys = new ArrayList<>();
+			String[] queryParamStrings = requestQueryParamString.split("\\&");
```
Comment (cameronbowie) [inline src/main/java/com/smartsheet/wiremock/extensions/ApiScenarioTransformer.java]: This too?

File: src/main/java/com/smartsheet/wiremock/extensions/ApiScenarioTransformer.java (hunk 1 above)
Comment (stollgr) [reply src/main/java/com/smartsheet/wiremock/extensions/ApiScenarioTransformer.java]: good catch. I thought that was where i put it.

**Pull request merged**